docs/NNNN-direct-clean-storage-layout.md

```diff
@@ -0,0 +1,99 @@
+# OCFL Community Extension NNNN: Direct Clean Storage Layout
+
+* **Extension Name:** NNNN-direct-clean-storage-layout
+* **Authors:** Jürgen Enge, Basel
+* **Minimum OCFL Version:** 1.0
+* **OCFL Community Extensions Version:** 1.0
+* **Obsoletes:** n/a
+* **Obsoleted by:** n/a
+
+## Overview
+
+This object/storage root extension maps content file names or OCFL object identifiers by replacing "dangerous characters" from file and folder names.
+
+The functionality derived from David A. Wheeler - "Fixing Unix/Linux/POSIX Filenames" (https://www.dwheeler.com/essays/fixing-unix-linux-filenames.html)
```

Comment:
Correct me if I'm wrong, but I don't believe that the algorithm that you're proposing here was actually described in this document. I think these are the rules proposed at the button of the article, yes?

If so, those rules were describing characters to _forbid_, and here you're attempting to sanitize them. A few paragraphs up, there's a section title "Make it possible for system administrators to force “bad” filenames to be automatically renamed when returned to userspace" that is the articles attempt to describe a sanitation process. The process they describe is essentially percent-encoding but with `=` instead of `%`.

The problem I have with the approach described here is that because it's simply removing or replacing characters, it's possible for you to construct conflicting paths. That is to say, the algorithm does not uniquely map inputs to outputs, which means that you could have storage path or content path collisions.

Perhaps you are comfortable with this for your use case, which is fine. However, I think that you should include a warning that this algorithm has limitations, and should only be used when you know it will not be an issue for your dataset.